ShelterModule/Models/Announcements/AnnouncementCreationRequest.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.ComponentModel.DataAnnotations;
+using ShelterModule.Models.Pets;
+
+namespace ShelterModule.Models.Announcements;
+
+public sealed class AnnouncementCreationRequest
+{
+    [Required]
+    public string Title { get; init; } = null!;
+
+    [Required]
+    public string Description { get; init; } = null!;
+
+    [Required]
+    public Guid ShelterId { get; init; }
+
+    public Guid? PetId { get; init; }
+
+    public PetUpsertRequest? PetRequest { get; init; }
```

Comment:
For the request to be valid, we need to either `PetId` of `PetRequest` to not be `null`. We can validate that explicitly in controller, but it's better to instead do this here. This class can implement `IValidatableObject`, which has a `Vallidate()` method, that will be called when the request is validated, and whose errors will be returned if validation fails.

Also, we need to check if the `PetRequest` object is valid, since validation attributes are not validated recursively by default. We can also do that in `Validate()` method.